crates/gitbutler-core/src/types/tagged_string.rs

```diff
@@ -0,0 +1,60 @@
+use std::{fmt, marker::PhantomData, ops::Deref};
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// Tagged string is designed to clarify the purpose of strings when used as a return type
+pub struct TaggedString<T>(String, PhantomData<T>);
```

Comment:
I recommend making this a `BString` as reference names for instance aren't limited to UTF8, despite having a limited character set. When working with Git idea that UTF8 is everywhere should be avoided in new code.

Reply:
I'll make sure to replace any instances with a BString, thanks for letting me know about that 👍